### PR TITLE
Add @pyohannes as owner of EC2/ECS/EKS AWS detectors

### LINKED ISSUE #5542: Deprecate and remove go.opentelemetry.io/otel/detectors/aws/ec2
This module has been identified to not have an [owner](https://github.com/open-telemetry/opentelemetry-go-contrib/blob/5d8c6d20b1852e47e15b989abe813c20cb935e9b/CODEOWNERS). Based on the [project's ownership policy](https://github.com/open-telemetry/opentelemetry-go-contrib/issues/2834#issuecomment-1282698918), this module will be deprecated and then removed.

### How to keep this module

For this module to continue in this repository, it needs a sponsor.

If you would like to sponsor this module and become an owner, please comment in this issue about your desire. As an owner you will assume the following responsibilities:

- You will need to [be a member of the OpenTelemetry organization](https://github.com/open-telemetry/community/blob/main/community-membership.md#member) and maintain that membership.
  - If you are not an existing member, we can add you to the community. This is not a disqualification, but something you will need in the process of becoming a sponsor.
- You will be responsible for keeping up with the instrumented library. Any "upstream" changes that impact this module need to be proactively handle by you.
- You will be expected to review any Pull Requests or Issues created that relate to this module.
- You will be responsible for the stability and versioning compliance of the module.

You will need to have a good working knowledge of the code this module is instrumenting and, ideally, familiarity with the existing module code.

### How this module will be removed

This module is in the process of being deprecated. After that deprecation notice has been published, we will wait 3 months or 2 two releases (whichever is the longer time period). After that time period, this module will be removed from this repository and no more versions of the module will be published.

#### Resurrection

If a sponsor is found after the module has been deprecated or removed, these operations can be reversed (i.e. coded added back, deprecation notice removed).

### LINKED ISSUE #5543: Deprecate and remove go.opentelemetry.io/otel/detectors/aws/ecs
This module has been identified to not have an [owner](https://github.com/open-telemetry/opentelemetry-go-contrib/blob/5d8c6d20b1852e47e15b989abe813c20cb935e9b/CODEOWNERS). Based on the [project's ownership policy](https://github.com/open-telemetry/opentelemetry-go-contrib/issues/2834#issuecomment-1282698918), this module will be deprecated and then removed.

### How to keep this module

For this module to continue in this repository, it needs a sponsor.

If you would like to sponsor this module and become an owner, please comment in this issue about your desire. As an owner you will assume the following responsibilities:

- You will need to [be a member of the OpenTelemetry organization](https://github.com/open-telemetry/community/blob/main/community-membership.md#member) and maintain that membership.
  - If you are not an existing member, we can add you to the community. This is not a disqualification, but something you will need in the process of becoming a sponsor.
- You will be responsible for keeping up with the instrumented library. Any "upstream" changes that impact this module need to be proactively handle by you.
- You will be expected to review any Pull Requests or Issues created that relate to this module.
- You will be responsible for the stability and versioning compliance of the module.

You will need to have a good working knowledge of the code this module is instrumenting and, ideally, familiarity with the existing module code.

### How this module will be removed

This module is in the process of being deprecated. After that deprecation notice has been published, we will wait 3 months or 2 two releases (whichever is the longer time period). After that time period, this module will be removed from this repository and no more versions of the module will be published.

#### Resurrection

If a sponsor is found after the module has been deprecated or removed, these operations can be reversed (i.e. coded added back, deprecation notice removed).

### LINKED ISSUE #5544: Deprecate and remove go.opentelemetry.io/otel/detectors/aws/eks
This module has been identified to not have an [owner](https://github.com/open-telemetry/opentelemetry-go-contrib/blob/5d8c6d20b1852e47e15b989abe813c20cb935e9b/CODEOWNERS). Based on the [project's ownership policy](https://github.com/open-telemetry/opentelemetry-go-contrib/issues/2834#issuecomment-1282698918), this module will be deprecated and then removed.

### How to keep this module

For this module to continue in this repository, it needs a sponsor.

If you would like to sponsor this module and become an owner, please comment in this issue about your desire. As an owner you will assume the following responsibilities:

- You will need to [be a member of the OpenTelemetry organization](https://github.com/open-telemetry/community/blob/main/community-membership.md#member) and maintain that membership.
  - If you are not an existing member, we can add you to the community. This is not a disqualification, but something you will need in the process of becoming a sponsor.
- You will be responsible for keeping up with the instrumented library. Any "upstream" changes that impact this module need to be proactively handle by you.
- You will be expected to review any Pull Requests or Issues created that relate to this module.
- You will be responsible for the stability and versioning compliance of the module.

You will need to have a good working knowledge of the code this module is instrumenting and, ideally, familiarity with the existing module code.

### How this module will be removed

This module is in the process of being deprecated. After that deprecation notice has been published, we will wait 3 months or 2 two releases (whichever is the longer time period). After that time period, this module will be removed from this repository and no more versions of the module will be published.

#### Resurrection

If a sponsor is found after the module has been deprecated or removed, these operations can be reversed (i.e. coded added back, deprecation notice removed).

### PR DESCRIPTION
**Pull request by MrAlias**
Closes #5542 
Closes #5543 
Closes #5544 

@pyohannes please be sure to review our [CODEOWNER policy](https://github.com/open-telemetry/opentelemetry-go-contrib/blob/main/CONTRIBUTING.md#code-owners). If you are okay with accepting the responsibilities, please approve this PR.

### CODE REVIEW

**Review by dmathieu — APPROVED** [review]

**Review by pellared — APPROVED** [review]

**Comment by pellared** [comment]
Let's not merge it until @pyohannes approves.

**Review by pyohannes — APPROVED** [review]

**Pull request merged**
